src/topology/algebra/constructions.lean

```diff
@@ -115,4 +118,37 @@ by simp only [inducing_embed_product.continuous_iff, embed_product_apply, (∘),
 @[to_additive] lemma continuous_coe_inv : continuous (λ u, ↑u⁻¹ : Mˣ → M) :=
 (units.continuous_iff.1 continuous_id).2
 
+/-- The inverse map `units.inv` is continuous with respect to the topology induced by
+  `units.coe_hom` on units.
+
+  Note this is not the topology used by default in mathlib.-/
+@[continuity]
+lemma induced_top_cont_inv {X : Type*} [topological_space X] [discrete_topology X] [monoid X] :
+  @continuous _ _ (topological_space.induced (units.coe_hom X) infer_instance) _
+  (units.inv : Xˣ → X) :=
+begin
+  convert continuous_of_discrete_topology,
+  refine @inducing.discrete_topology Xˣ X (topological_space.induced
+    (units.coe_hom X) infer_instance) _ _ (units.coe_hom _)
+    (λ a b h, units.eq_iff.1 h) rfl,
+-- Alternate proof :
+--  refine @embedding.discrete_topology (zmod n)ˣ (zmod n) (topological_space.induced
+--    (units.coe_hom (zmod n)) infer_instance) _ _ (units.coe_hom _) _,
+--  constructor,
+--  { constructor, refl, },
+--  { refine λ a b h, units.eq_iff.1 h, }, }
+end
+
+lemma discrete_topology_of_discrete {X : Type*} [_root_.topological_space X] [discrete_topology X]
```

Comment:
Can you make that an instance and make sure the name is `units.discrete_topology` (probably you just need to remove the explicit name)?